packages/suite/src/components/suite/HomescreenGallery.tsx

```diff
@@ -43,22 +43,28 @@ type HomescreenGalleryProps = {
     onConfirm?: () => void;
 };
 
-const HomescreenGallery = ({ device, onConfirm }: HomescreenGalleryProps) => {
+export const HomescreenGallery = ({ device, onConfirm }: HomescreenGalleryProps) => {
     const { applySettings } = useActions({ applySettings: deviceSettingsActions.applySettings });
 
     const deviceModel = getDeviceModel(device);
 
     if (!deviceModel) return null;
 
-    const setHomescreen = (image: AnyImageName) => {
-        const element = document.getElementById(image);
-        if (element instanceof HTMLImageElement) {
-            const hex = elementToHomescreen(element, deviceModel);
-            applySettings({ homescreen: hex });
-            if (onConfirm) {
-                onConfirm();
-            }
+    const setHomescreen = async (imagePath: string, image: AnyImageName) => {
```

Comment:
It should be somehow disabled when device is locked. Otherwise it's easy to get multiple "Device call in progress" notifications by clicking repeatedly.
<img width="542" alt="image" src="https://user-images.githubusercontent.com/3729633/225012205-d811ba77-ab08-4611-b241-865e4cd5bb0f.png">

Reply:
I dont have this issue. When I choose a homescreen and device is locked. It first prompts device to unlock and then to confirm image with this screen in Suite.
<img width="456" alt="Screenshot 2023-03-27 at 11 55 30" src="https://user-images.githubusercontent.com/33235762/227922317-04d0f546-06c2-4af8-be1e-01cea6c79425.png">

Reply:
try to click on one of the homescreen from gallery rapidly - multiple times

Reply:
this is what I meant: d336e0e08b40e7099823f4a11a58b3b4f649e6fc